lib/palindrome_check.rb

```diff
@@ -1,5 +1,28 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil
+    return false
+  end
+  i = 0
+  j = my_phrase.length - 1
+  while i < j
+    if my_phrase[i] == " " || my_phrase[j] == " "
+      until my_phrase[i] != " "
```

Comment:
In the two nested inner loops, continue to check and confirm that i is less than j before comparing the character at the index i (or j)  with a white space.